### PR TITLE
[Distributed] Add `repr` methods for `ParallelStyle`s

### LINKED ISSUE #149470: `ParallelStyle`s (ColwiseParallel, etc) do not have a `__repr__()`
### 🚀 The feature, motivation and pitch

I'm writing a TP plan for a new model, and it's not possible to print the dict to copy/paste it. It's making debugging the parallelism strategies from torchtitan much harder. `Placement` objects already have a `__repr__`, so this should be easy to support.

### Alternatives

_No response_

### Additional context

_No response_

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### PR DESCRIPTION
**Pull request by shink**
Fixes #149470

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

**Review by apaz-cli — COMMENTED** [review]
You can't pass `src_data_rank` to the constructor, so I wonder if it should be in the `__repr__()`. Typically python says no, but it is potentially good info to have. I'll leave it up to the judgement of others.

Other than that, lgtm.

File: torch/distributed/tensor/parallel/style.py [hunk 1]
```diff
@@ -166,6 +166,15 @@ def _apply(self, module: nn.Module, device_mesh: DeviceMesh) -> nn.Module:
             ),
         )
 
+    def __repr__(self) -> str:
+        tmpstr = self.__class__.__name__ + "("
+        tmpstr += f"src_data_rank={self.src_data_rank}, "
```
Comment (wanchaol) [inline torch/distributed/tensor/parallel/style.py:171]: I think we should remove `src_data_rank` from `__repr__` (including all the following changes). Python's `__repr__` ideally can be used to recreate the object using eval()

File: torch/distributed/tensor/parallel/style.py [hunk 2]
```diff
@@ -398,6 +416,14 @@ def _apply(self, module: nn.Module, device_mesh: DeviceMesh) -> nn.Module:
             partial(self._prepare_output_fn, self.use_local_output),
         )
 
+    def __repr__(self) -> str:
+        tmpstr = self.__class__.__name__ + "("
+        tmpstr += f"src_data_rank={self.src_data_rank}, "
+        tmpstr += f"sequence_sharding={self.sequence_sharding}, "
```
Comment (wanchaol) [inline torch/distributed/tensor/parallel/style.py:422]: should be `sequence_dim=self.sequence_sharding.dim`

**Review by wanchaol — COMMENTED** [review]
Please see comments inlined

File: torch/distributed/tensor/parallel/style.py (hunk 1 above)
Comment (shink) [reply torch/distributed/tensor/parallel/style.py]: Thanks for your review! Have made changes and please take a look, thanks!

**Comment by wconstab** [comment]
Can you paste outputs from the __repr__ to confirm they are looking good? Thanks for the fix!

**Comment by shink** [comment]
@wconstab Thanks for your review! Here's a simple output. Does it make sense?

```python
def fn():
    from torch.distributed.tensor.parallel.style import (
        ColwiseParallel,
        PrepareModuleInput,
        PrepareModuleOutput,
        RowwiseParallel,
        SequenceParallel,
    )
    from torch.distributed.tensor.placement_types import Shard

    a = ColwiseParallel()
    b = PrepareModuleInput()
    c = PrepareModuleOutput(output_layouts=Shard(1), desired_output_layouts=Shard(2))
    d = RowwiseParallel()
    e = SequenceParallel()
    print(a)
    print(b)
    print(c)
    print(d)
    print(e)
```

Got:

```
ColwiseParallel(input_layouts=(Replicate(),), output_layouts=(Shard(dim=-1),), use_local_output=True)
PrepareModuleInput(input_layouts=None, desired_input_layouts=None, input_kwarg_layouts={}, desired_input_kwarg_layouts={}, use_local_output=False)
PrepareModuleOutput(output_layouts=(Shard(dim=1),), desired_output_layouts=(Shard(dim=2),), use_local_output=True)
RowwiseParallel(input_layouts=(Shard(dim=-1),), output_layouts=(Replicate(),), use_local_output=True)
SequenceParallel(sequence_dim=1, use_local_output=False)
```

**Review by wanchaol — APPROVED** [review]
thanks!
